tests/garage/tf/optimizers/test_conjugate_gradient_optimizer.py

```diff
@@ -0,0 +1,87 @@
+"""Tests for garage.tf.optimizers.conjugateGradientOptimizer"""
+import numpy as np
+import tensorflow as tf
+
+from garage.tf.optimizers.conjugate_gradient_optimizer import (
+    cg, FiniteDifferenceHvp, PerlmutterHvp)
+from garage.tf.policies import Policy
+
+
+class OneParamPolicy(Policy):
+    """Helper class for testing hvp classes"""
+
+    def __init__(self, name='OneParamPolicy'):
+        super().__init__(name, None)
+        with tf.compat.v1.variable_scope(self.name) as vs:
+            self._variable_scope = vs
+            _ = tf.Variable([0.])
+
+    def get_action(self, observation):
+        pass
+
+    def get_actions(self, observations):
+        pass
+
+
+class TestConjugateGradientOptimizer:
+    """Test class for ConjugateGradientOptimizer and HVP classes"""
+
+    def test_cg(self):
+        """Solve Ax = b using Conjugate gradient method."""
+        a = np.random.randn(5, 5)
+        a = a.T.dot(a)  # make sure a is positive semi-definite
+        b = np.random.randn(5)
+        x = cg(a.dot, b, cg_iters=5)
+        assert np.allclose(a.dot(x), b)
+
+
+class TestPermmutterHvp:
+    """Test class for PermmutterHvp"""
+
+    def test_perm_mutter_hvp(self):
+        """Test if hessian product calculations are correct"""
+        policy = OneParamPolicy()
+        x = policy.get_params()[0]
+        a_val = np.array([5.0])
```

Comment:
i would be more comfortable if you tested this on a range of cases.

at the very least, a 1x1 Hessian matrix doesn't test much because you don't even have off-diagonal elements. You would at least want a 2x2 matrix.